src/thermal.rs

```diff
@@ -0,0 +1,40 @@
+use super::Error;
+use std::fs::read_dir;
+
+const THERMAL_ZONE_DIR: &str = "/sys/class/thermal/";
+
+pub struct ThermalZone {
+    pub name: String,
+    pub location: String,
+    pub temp: i32,
+    pub enabled: bool,
+}
+
+pub trait Thermal {
+    fn update(&mut self) -> Result<(), Error>;
+}
+
+impl Default for ThermalZone {
+    fn default() -> Self {
+        ThermalZone {
+            name: "unknown".to_string(),
+            location: "unknown".to_string(),
+            temp: 0,
+            enabled: false,
+        }
+    }
+}
+
+pub fn read_thermal_zones() -> Vec<ThermalZone> {
+    let mut zones = Vec::<ThermalZone>::new();
+
+    for a in read_dir(THERMAL_ZONE_DIR).expect("Could not read thermal directory") {
```

Comment:
also fun fact, there are 19 thermal zones on my hexacore laptop, 11 thermal zones on my octacore laptop, and 2 on my dual-core laptop

Reply:
> prob wanna check the type first, since most thermal zones are just constants (there is no thermal zones for each individual cores, instead it is shown as one `x86_pkg_temp`)

I'm planning on converting type to an enum with a catch all type "other" for all undocumented thermal zones. That way we can filter out all virtual thermal zones

Reply:
> I'm planning on converting type to an enum with a catch all type "other" for all undocumented thermal zones. That way we can filter out all virtual thermal zones

This doesn't seem to be necessary.